src/transformers/generation/candidate_generator.py

```diff
@@ -225,7 +225,8 @@ def update_candidate_strategy(self, input_ids: torch.LongTensor, scores: torch.F
         # Adjust the max number of assistant tokens to use in the next iteration. This is a simple heuristic,
         # probably can be improved -- we want to balance the benefits of getting assistant tokens correct with the
         # cost of forecasting incorrect assistant tokens.
-        if self.assistant_model.generation_config.num_assistant_tokens_schedule == "heuristic":
+        if self.assistant_model.generation_config.num_assistant_tokens_schedule == "heuristic" or \
+            self.assistant_model.generation_config.num_assistant_tokens_schedule == "heuristic_transient":
```

Comment:
```suggestion
        if self.assistant_model.generation_config.num_assistant_tokens_schedule == "heuristic" or self.assistant_model.generation_config.num_assistant_tokens_schedule == "heuristic_transient":
```

Run `make fixup` in your terminal afterwards, and commit the result. Our CI will complain otherwise :)